### PR TITLE
Parent table with params

### PR DESCRIPTION
**Pull request by anncatton**
Adds parameters to list context
- adds back sorting, search and pagination
- fixes table sort. the api results were in the correct order, but the table logic was reversing the results, wheeee

### CODE REVIEW

File: src/components/global/hooks/useListContext.tsx [hunk 1]
```diff
@@ -99,49 +92,56 @@ export const ListProvider = ({
...
-      const newParams = {
-        ...currentListParams,
-        sortField: getInitialSortField(resourceName),
-      };
-      // TODO: there's still 2 api requests happening, because of params update i think. need to investigate
-      const data = await getList(newParams);
-      setListState(data);
-    }
-  }, [resourceName, currentListParams]);
+  const setListParams = useMemo(
```
Comment (ciaranschutte) [inline src/components/global/hooks/useListContext.tsx]: not sure why we're using `useMemo` here?

File: src/components/global/hooks/useDebounce.tsx [hunk 2]
```diff
@@ -0,0 +1,21 @@
+import { useState, useEffect } from 'react';
+
+// from https://usehooks.com/useDebounce/
+
+const useDebounce = <T,>(value: T, delay: number): T => {
```
Comment (ciaranschutte) [inline src/components/global/hooks/useDebounce.tsx:5]: <T,> ? weird. prettier checks out though

File: src/components/ListPane/ListPane.tsx [hunk 3]
```diff
@@ -148,7 +152,9 @@ const List = () => {
                 backgroundColor: 'transparent',
                 ...(listParams.sortOrder === 'DESC' && { color: theme.colors.primary_5 }),
               }}
-              // onClick={() => setListParams({ ...listParams, sortOrder: 'DESC' })}
+              onClick={() =>
+                setListParams({ sortOrder: listParams.sortOrder === 'ASC' ? 'DESC' : 'ASC' })
```
Comment (ciaranschutte) [inline src/components/ListPane/ListPane.tsx]: worth putting 'ASC' and 'DESC' into enum? used a fair bit

**Review by ciaranschutte — APPROVED** [review]
lgtm
one suggestion if you fancy

File: src/components/global/hooks/useDebounce.tsx (hunk 2 above)
Comment (anncatton) [reply src/components/global/hooks/useDebounce.tsx:5]: yeah i just lazily copied it...we have it in dac-ui too 🤷‍♀️

File: src/components/ListPane/ListPane.tsx (hunk 3 above)
Comment (anncatton) [reply src/components/ListPane/ListPane.tsx]: Funny I thought I did have one, and forgot to use it here...but will do!

Comment (anncatton) [reply src/components/ListPane/ListPane.tsx]: Nvmd i def do, it was just a type instead of an enum...will use it here!

**Pull request merged**